include/aws/common/condition_variable.h

```diff
@@ -33,6 +33,7 @@ struct aws_condition_variable {
 #else
     pthread_cond_t condition_handle;
 #endif
+    bool initialized;
```

Comment:
Isn't condition_handle != 0/NULL enough to flag init?

Reply:
On Windows, the AWS_CONDITION_VARIABLE_INIT macro zeros out the datastructure. So checking if it's zeroed will not tell you whether or not its been initialized.

For pthread, it's **probably** one of those "The spec doesn't say that 0 can't be valid" things. I'm just speculating there. I don't know where one goes to read the pthread standard. I dug into the headers on Mac and `PTHREAD_COND_INITIALIZER` makes it non-zero, but I can't say for sure on other platforms.